src/services/importTracker.ts

```diff
@@ -140,6 +146,20 @@ namespace ts.FindAllReferences {
             }
         }
 
+        function handleImportCall(importCall: ImportCall) {
+            const top = findAncestor(importCall, isAmbientModuleDeclaration) || importCall.getSourceFile();
+            const exported = isExported(importCall, top);
+            if (exported) addIndirectUsers(top);
+            else addIndirectUser(top);
+        }
+
+        function isExported(node: Node, top: SourceFileLike = node.getSourceFile()) {

```

Comment:
You can avoid a double walk up the tree by changing `top` to `stopAtAmbientModule: boolean = false` and `if (node === top)` to `if (stopAtAmbientModule && isAmbientModuleDeclaration(node))`